script/releases/Env.sol

```diff
@@ -0,0 +1,266 @@
+    }
+
+    function MAX_RETROACTIVE_LENGTH() internal view returns (uint32) {
+        return _envU32("REWARDS_COORDINATOR_MAX_RETROACTIVE_LENGTH");
+    }
+
+    function MAX_FUTURE_LENGTH() internal view returns (uint32) {
+        return _envU32("REWARDS_COORDINATOR_MAX_FUTURE_LENGTH");
+    }
+
+    function GENESIS_REWARDS_TIMESTAMP() internal view returns (uint32) {
+        return _envU32("REWARDS_COORDINATOR_GENESIS_REWARDS_TIMESTAMP");
+    }
+
+    /**
+     * core/
+     */
+
+    function allocationManager(DeployedProxy) internal view returns (AllocationManager) {
+        return AllocationManager(_deployedProxy(type(AllocationManager).name));
+    }
+
+    function allocationManager(DeployedImpl) internal view returns (AllocationManager) {
+        return AllocationManager(_deployedImpl(type(AllocationManager).name));
+    }
+
+    function avsDirectory(DeployedProxy) internal view returns (AVSDirectory) {
+        return AVSDirectory(_deployedProxy(type(AVSDirectory).name));
+    }
+
+    function avsDirectory(DeployedImpl) internal view returns (AVSDirectory) {
+        return AVSDirectory(_deployedImpl(type(AVSDirectory).name));
+    }
+
+    function delegationManager(DeployedProxy) internal view returns (DelegationManager) {
+        return DelegationManager(_deployedProxy(type(DelegationManager).name));
+    }
+
+    function delegationManager(DeployedImpl) internal view returns (DelegationManager) {
+        return DelegationManager(_deployedImpl(type(DelegationManager).name)); 
+    }
+
+    function rewardsCoordinator(DeployedProxy) internal view returns (RewardsCoordinator) {
+        return RewardsCoordinator(_deployedProxy(type(RewardsCoordinator).name));
+    }
+
+    function rewardsCoordinator(DeployedImpl) internal view returns (RewardsCoordinator) {
+        return RewardsCoordinator(_deployedImpl(type(RewardsCoordinator).name)); 
+    }
+
+    function strategyManager(DeployedProxy) internal view returns (StrategyManager) {
+        return StrategyManager(_deployedProxy(type(StrategyManager).name));
+    }
+
+    function strategyManager(DeployedImpl) internal view returns (StrategyManager) {
+        return StrategyManager(_deployedImpl(type(StrategyManager).name)); 
+    }
+
+    /**
+     * permissions/
+     */
+
+    function pauserRegistry(DeployedImpl) internal view returns (PauserRegistry) {
+        return PauserRegistry(_deployedImpl(type(PauserRegistry).name));
+    }
+
+    function permissionController(DeployedProxy) internal view returns (PermissionController) {
+        return PermissionController(_deployedProxy(type(PermissionController).name));
+    }
+
+    function permissionController(DeployedImpl) internal view returns (PermissionController) {
+        return PermissionController(_deployedImpl(type(PermissionController).name)); 
+    }
+
+    /**
+     * pods/
+     */
+
+    function eigenPod(DeployedProxy) internal view returns (EigenPod) {
```

Comment:
hm. this is gross. makes it weird to treat a beacon the same way as TUPs...

I need to read thru zeus

Reply:
There's room to make this `Env.beacon.eigenPod` instead, and treat the two separately. I'm down.